docs/examples/workflows/experimental/new_dag_decorator_params.md

```diff
@@ -0,0 +1,187 @@
+# New Dag Decorator Params
+
+
+
+
+
+
+=== "Hera"
+
+    ```python linenums="1"
+    from pydantic import BaseModel
+    from typing_extensions import Annotated
+
+    from hera.shared import global_config
+    from hera.workflows import Input, Output, Parameter, Workflow
+
+    global_config.experimental_features["script_annotations"] = True
+    global_config.experimental_features["script_pydantic_io"] = True
+
+
+    w = Workflow(generate_name="my-workflow-")
+
+
+    class SetupConfig(BaseModel):
+        a_param: str
+
+
+    class SetupOutput(Output):
+        environment_parameter: str
+        an_annotated_parameter: Annotated[int, Parameter(name="dummy-param")]  # use an annotated non-str
+        setup_config: Annotated[SetupConfig, Parameter(name="setup-config")]  # use a pydantic BaseModel
+
+
+    @w.script()
+    def setup() -> SetupOutput:
+        return SetupOutput(
+            environment_parameter="linux",
+            an_annotated_parameter=42,
+            setup_config=SetupConfig(a_param="test"),
+            result="Setting things up",
+        )
+
+
+    class ConcatInput(Input):
+        word_a: Annotated[str, Parameter(name="word_a", default="")]
+        word_b: str
+
+
+    @w.script()
+    def concat(concat_input: ConcatInput) -> Output:
+        return Output(result=f"{concat_input.word_a} {concat_input.word_b}")
+
+
+    class WorkerInput(Input):
+        value_a: str = "my default"
+        value_b: str
+        an_int_value: int = 42
+
+
+    class WorkerOutput(Output):
+        value: str
+
+
+    @w.set_entrypoint
+    @w.dag()
+    def worker(worker_input: WorkerInput) -> WorkerOutput:
+        setup_task = setup()
```

Comment:
Thanks - this uncovered a bug in `_get_as_arguments`, fixed in latest push, where we need to serialize the value from the model dump:
```py
            # The value may be a static value (of any time) if it has a default value, so we need to serialize it
            # If it is a templated string, it will be unaffected as `"{{mystr}}" == serialize("{{mystr}}")``
            templated_value = serialize(self_dict[field])
```